configure.ac

```diff
@@ -2118,65 +2118,48 @@ if test "X$MULTITHREAD" = "Xyes"; then
 
     case "$host_cpu-$host_vendor-$host_os" in
       *linux*)
-	# Replace C99 with C11 standard to support atomic datatype
-	H5_CFLAGS=$(echo $H5_CFLAGS | sed -e "s/-std=c99/-std=c11/g")
+        HAVE_STDATOMIC_H="yes"
 
-        # Remove the flag -Wc99-c11-compat to supress compiler warnings
-        # No need of the warnings for the features not in C99 because we
-        # have to use C11
-        H5_CFLAGS=$(echo $H5_CFLAGS | sed -e "s/-Wc99-c11-compat//g")
+        # Check the presence of stdatomic.h for atomic data type
+        AC_CHECK_HEADERS([stdatomic.h],,[HAVE_STDATOMIC_H="no"])
 
-	# Propagate the change to CFLAGS, too
-	CFLAGS="$H5_CFLAGS $AM_CFLAGS $CFLAGS"
```

Comment:
Remove the above, now outdated, lines that were trying to adjust the C standard to C11 and remove the `-Wc99-c11-compat` flag. The same in the 'apple' section below. The rest of the lines were automatically converted from tabs to spaces.